src/resource/saveSolidDatasetInContainer.ts

```diff
@@ -0,0 +1,80 @@
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import {
+  UrlString,
+  saveSolidDatasetInContainer as coreSaveSolidDatasetInContainer,
+  SolidDataset,
+  WithResourceInfo,
+} from "@inrupt/solid-client";
+import { VerifiableCredential } from "@inrupt/solid-client-vc";
+import { fetchWithVc } from "../fetch";
+import { FetchOptions } from "../type/FetchOptions";
+
+/**
+ * Given a SolidDataset, store it in a Solid Pod as a new Resource inside a Container.
+ *
+ * The Container at the given URL should already exist; if it does not, you can initialise it first
+ * using [[createContainerAt]], or directly save the SolidDataset at the desired location using
+ * [[saveSolidDatasetAt]].
+ *
+ * This function is primarily useful if the current user does not have access to change existing files in
+ * a Container, but is allowed to add new files; in other words, they have Append, but not Write
+ * access to a Container. This is useful in situations where someone wants to allow others to,
+ * for example, send notifications to their Pod, but not to view or delete existing notifications.
+ * You can pass a suggestion for the new Resource's name, but the server may decide to give it
+ * another name — for example, if a Resource with that name already exists inside the given
+ * Container.
+ * If the user does have access to write directly to a given location, [[saveSolidDatasetAt]]
+ * will do the job just fine, and does not require the parent Container to exist in advance.
+ *
+ * @see [@inrupt/solid-client's
+ * saveSolidDatasetInContainer](https://docs.inrupt.com/developer-tools/api/javascript/solid-client/modules/resource_solidDataset.html#savesoliddatasetincontainer)
+ *
+ *
+ * @param containerUrl URL of the Container in which to create a new Resource.
+ * @param solidDataset The [[SolidDataset]] to save to a new Resource in the given Container.
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @returns A Promise resolving to a [[SolidDataset]] containing the saved data. The Promise rejects if the save failed.
+ */
+
+export async function saveSolidDatasetInContainer(
+  containerUrl: UrlString,
```

Comment:
Note: this function in solid-client accepted a containerUrl with type UrlString or Url. I had to remove the Url type here so that it would work with the fetchWithVC function which only accepts a UrlString.

Reply:
`Url | UrlString` is tech debt, basically they both resolve down to `string` at the end of the day.